pennylane/templates/layers/particle_conserving_u1.py

```diff
@@ -0,0 +1,253 @@
+    elementary gates. The Pauli matrix :math:`\sigma_z` and :math:`R(0, 2 \theta, 0)`
+    apply the Pauli Z operator and an arbitrary rotation on the qubit ``n``,
+
+    |
+
+    .. figure:: ../../_static/templates/layers/u1_decomposition.png
+        :align: center
+        :width: 80%
+        :target: javascript:void(0);
+
+    |
+
+    :math:`U_A(\phi)` is the unitary matrix
+
+    .. math::
+
+        U_A(\phi) = \left(\begin{array}{cc} 0 & e^{-i\phi} \\ e^{-i\phi} & 0 \\ \end{array}\right)
+
+    acting controlledly on the state of qubit ``m`` which is further decomposed in terms of the
+    `quantum operations <https://pennylane.readthedocs.io/en/stable/introduction/operations.html>`_
+    supported by Pennylane,
+
+    |
+
+    .. figure:: ../../_static/templates/layers/ua_decomposition.png
+        :align: center
+        :width: 70%
+        :target: javascript:void(0);
+
+    |
+
+    where,
+
+    |
+
+    .. figure:: ../../_static/templates/layers/phaseshift_decomposition.png
+        :align: center
+        :width: 65%
+        :target: javascript:void(0);
+
+    |
+
+    The quantum circuits above decomposing the unitaries :math:`U_{1,\mathrm{ex}}(\phi, \theta)`
+    and :math:`U_A(\phi)` are implemented by the :func:`~.u1_ex_gate` and :func:`~.decompose_ua`
+    functions, respectively.
+
+    Args:
+        weights (array[float]): Array of weights of shape ``(D, M, 2)``.
+            ``D`` is the number of entangler block layers and :math:`M=N-1`
+            is the number of exchange gates :math:`U_{1,\mathrm{ex}}` per layer.
+        wires (Iterable or Wires): Wires that the template acts on. Accepts an iterable of numbers
+            or strings, or a Wires object.
+        init_state (array[int]): Length ``len(wires)`` occupation-number vector representing the
+            HF state. ``init_state`` is used to initialize the wires.
+
+    Raises:
+        ValueError: if inputs do not have the correct format
+
+    .. UsageDetails::
+
+        Notice that:
+
+        #. The number of wires :math:`N` has to be equal to the number of
+           spin orbitals included in the active space.
+
+        #. The number of trainable parameters scales linearly with the number of layers as
+           :math:`2D(N-1)`.
+
+        An example of how to use this template is shown below:
+
+        .. code-block:: python
+
+            import pennylane as qml
+            from pennylane.templates import ParticleConservingU1
+            from functools import partial
+
+            # Build the electronic Hamiltonian
+            h, qubits = qml.qchem.molecular_hamiltonian("h2", "h2.xyz")
+
+            # Define the HF state
+            ref_state = qml.qchem.hf_state(electrons=2, qubits)
+
+            # Define the device
+            dev = qml.device('default.qubit', wires=qubits)
+
+            # Define the ansatz
+            ansatz = partial(ParticleConservingU1, init_state=ref_state)
```

Comment:
I remember the use of `partial` being suggested by Josh previously. Is this how all templates are called?

Reply:
Not all, but for QChem templates that typically require keyword arguments, seems convenient